File: flutter_embed_unity_2022_3_ios/ios/Classes/View/IUnityViewStackable.swift
//
//  IUnityViewStack.swift
//  flutter_embed_unity_ios
//
//  Created by James Allen on 28/08/2023.
//

import Foundation
import UnityFramework

protocol IUnityViewStackable {
    func detachUnity()
    func attachUnity(_ unityPlayerSingleton: UnityFramework)
}

File: flutter_embed_unity_6000_android/android/src/main/kotlin/com/learntoflutter/flutter_embed_unity_android/platformView/IUnityViewStackable.kt
package com.learntoflutter.flutter_embed_unity_android.platformView

import com.learntoflutter.flutter_embed_unity_android.unity.UnityPlayerSingleton

interface IUnityViewStackable {
    fun attachUnity(unityPlayerSingleton: UnityPlayerSingleton)
    fun detachUnity()
    var onDispose: (() -> Unit)?
}